library/ssl_cli.c

```diff
@@ -2021,14 +2027,20 @@ static int ssl_check_server_ecdh_params( const mbedtls_ssl_context *ssl )
     MBEDTLS_SSL_DEBUG_MSG( 2, ( "ECDH curve: %s", curve_info->name ) );
 
 #if defined(MBEDTLS_ECP_C)
-    if( mbedtls_ssl_check_curve( ssl, ssl->handshake->ecdh_ctx.grp.id ) != 0 )
+    if( mbedtls_ssl_check_curve( ssl, grp_id ) != 0 )
 #else
     if( ssl->handshake->ecdh_ctx.grp.nbits < 163 ||
         ssl->handshake->ecdh_ctx.grp.nbits > 521 )
 #endif
         return( -1 );
 
+#if defined(MBEDTLS_ECDH_LEGACY_CONTEXT)
```

Comment:
This 7-line pattern (`#if LEGACY_CONTEXT`, `DEBUG`, `#ELSE`, `if`, `DEBUG`, continuation, `#endif`) is repeated many times and is due to evolve to support other context variants. It should be an auxiliary function or macro.